internal/admission/handler.go

```diff
@@ -0,0 +1,65 @@
+package admission
+
+import (
+	"context"
+	"fmt"
+
+	aes "github.com/argoproj/argo-events/pkg/apis/eventsource"
+	as "github.com/argoproj/argo-events/pkg/apis/sensor"
+	event "github.com/kanopy-platform/argoslower/internal/admission/eventsource"
+	sensor "github.com/kanopy-platform/argoslower/internal/admission/sensor"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+type RoutingHandler struct {
+	sensorHandler      *sensor.Handler
+	eventSourceHandler *event.Handler
+}
+
+func NewRoutingHandler(sh *sensor.Handler, es *event.Handler) *RoutingHandler {
+	return &RoutingHandler{
+		sensorHandler:      sh,
+		eventSourceHandler: es,
+	}
+}
+
+func (h *RoutingHandler) Handle(ctx context.Context, req admission.Request) admission.Response {
+	kind := req.RequestKind
+	if kind == nil {
+		kind = &req.Kind
+	}
+
+	switch {
+	case kind.Kind == aes.Kind && h.eventSourceHandler != nil:
+		return h.eventSourceHandler.Handle(ctx, req)
+	case kind.Kind == as.Kind && h.sensorHandler != nil:
+		return h.sensorHandler.Handle(ctx, req)
+	default:
+		return admission.Denied(fmt.Sprintf("Kind %s not supported by controller", kind.Kind))
+	}
+
+}
+
+func (h *RoutingHandler) InjectDecoder(decoder *admission.Decoder) error {
+	if h.sensorHandler != nil {
+		err := h.sensorHandler.InjectDecoder(decoder)
+		if err != nil {
+			return err
+		}
+	}
+	if h.eventSourceHandler != nil {
+		err := h.eventSourceHandler.InjectDecoder(decoder)
+		if err != nil {
+			return err
+		}
+
+	}
+
+	return nil
+}
+
+func (h *RoutingHandler) SetupWithManager(m manager.Manager) {
+	m.GetWebhookServer().Register("/mutate", &webhook.Admission{Handler: h})
```

Comment:
looks good, should the [sensor.Handler](https://github.com/kanopy-platform/argoslower/blob/main/internal/admission/sensor/handler.go#L32) be moved to `/mutate/sensor`?

I know it's not being used, but to avoid confusion and a panic if someone calls SetupWithManager on it

Reply:
I was keeping the original behavior here.  I wonder if having this function take a string arg would make this easier in the future.

Reply:
Not sure I follow.  There is only a single path "/mutate" for the webhook.   The routing is internal based on the `kind` in the request. 

A string arg seems unnecessary since to add a new `kind` is simply extending the the `switch...case` statement above.

Reply:
It was a comment on the unused SetupWithManager that still exists for [sensor](https://github.com/kanopy-platform/argoslower/blob/main/internal/admission/sensor/handler.go#L32) and [eventsource](https://github.com/kanopy-platform/argoslower/blob/main/internal/admission/eventsource/handler.go#L43)

Calling .Register() on same path results in panic.
```
// Register marks the given webhook as being served at the given path.
// It panics if two hooks are registered on the same path.
```

But since they're unused, it's not an issue.